### PR TITLE
Implement server profile GraphQL queries

### PR DESCRIPTION
**Pull request by mkpankov**
Running creation mutation:
image

Results in DB:
```
chroma=# select * from chroma_core_serverprofile;
          name          |                ui_name                |                                        ui_description                                        | managed | worker | user_selectable | initial_state | ntp | corosync | corosync2 | pacemaker | default 
------------------------+---------------------------------------+----------------------------------------------------------------------------------------------+---------+--------+-----------------+---------------+-----+----------+-----------+-----------+---------
 robinhood_server       | Robinhood Policy Engine Server (MLNX) | A server running the Robinhood Policy Engine, with Mellanox IB support                       | t       | t      | t               | working       | t   | f        | f         | f         | f
 base_managed_rh7       | Managed Storage Server                | A storage server suitable for creating new HA-enabled filesystem targets                     | t       | f      | t               | managed       | t   | f        | t         | t         | f
 foo                    | foo                                   | fooDescription                                                                               | f       | f      | t               | started       | t   | t        | t         | t         | f
 base_managed_ib_rh7    | Managed Storage Server (MLNX)         | A storage server suitable for creating new HA-enabled filesystem targets using Mellanox OFED | t       | f      | t               | managed       | t   | f        | t         | t         | f
 base_managed_patchless | Managed Storage Server (patchless)    | A storage server suitable for creating new HA-enabled filesystem targets                     | t       | f      | t               | managed       | t   | f        | t         | t         | f
 base_monitored         | Storage Server                        | A storage server for monitoring and management of pre-existing resources                     | f       | f      | t               | monitored     | f   | f        | f         | f         | f
 default                | Unconfigured Server                   | An unconfigured server.                                                                      | f       | f      | f               | unconfigured  | f   | f        | f         | f         | f
 default_baseless       | Unconfigured Baseless Server          | An unconfigured server without a base.repo.                                                  | f       | f      | f               | unconfigured  | f   | f        | f         | f         | f
 posix_copytool_worker  | POSIX HSM Agent Node (MLNX)           | An HSM agent node using the POSIX copytool, with Mellanox IB support                         | t       | t      | t               | working       | t   | f        | f         | f         | f
(9 rows)

(reverse-i-search)`': ^C
chroma=# select * from chroma_core_serverprofile_repolist;
 id |    serverprofile_id    |         repo_id         
----+------------------------+-------------------------
  1 | base_managed_rh7       | base
  2 | base_managed_rh7       | lustre-server
  3 | base_managed_ib_rh7    | base
  4 | base_managed_ib_rh7    | lustre-server-mlnx
  5 | base_managed_patchless | base
  6 | base_managed_patchless | lustre-server-patchless
  7 | base_monitored         | base
  8 | default                | base
  9 | posix_copytool_worker  | base
 12 | robinhood_server       | base
 76 | posix_copytool_worker  | lustre-client-mlnx
 78 | robinhood_server       | lustre-client-mlnx
 79 | foo                    | base
 80 | foo                    | lustre-server
(14 rows)

chroma=# select * from chroma_core_serverprofilepackage;
 id |         package_name         |   server_profile_id    
----+------------------------------+------------------------
  1 | python2-iml-agent-management | base_managed_rh7
  2 | kernel-devel-lustre          | base_managed_rh7
  3 | pcs                          | base_managed_rh7
  4 | fence-agents                 | base_managed_rh7
  5 | fence-agents-virsh           | base_managed_rh7
  6 | lustre-resource-agents       | base_managed_rh7
  7 | lustre-ldiskfs-zfs           | base_managed_rh7
  8 | python2-iml-agent-management | base_managed_ib_rh7
  9 | kernel-devel-lustre          | base_managed_ib_rh7
 10 | pcs                          | base_managed_ib_rh7
 11 | fence-agents                 | base_managed_ib_rh7
 12 | fence-agents-virsh           | base_managed_ib_rh7
 13 | mlnx-ofa_kernel-devel        | base_managed_ib_rh7
 14 | kmod-mlnx-ofa_kernel         | base_managed_ib_rh7
 15 | lustre-resource-agents       | base_managed_ib_rh7
 16 | lustre-ldiskfs-zfs           | base_managed_ib_rh7
 17 | python2-iml-agent-management | base_managed_patchless
 18 | pcs                          | base_managed_patchless
 19 | fence-agents                 | base_managed_patchless
 20 | fence-agents-virsh           | base_managed_patchless
 21 | lustre-resource-agents       | base_managed_patchless
 22 | lustre-ldiskfs-zfs-patchless | base_managed_patchless
 23 | python2-iml-agent-management | base_monitored
 24 | python2-iml-agent-management | posix_copytool_worker
 25 | lustre-client                | posix_copytool_worker
 28 | python2-iml-agent-management | robinhood_server
 29 | lustre-client                | robinhood_server
 30 | robinhood-lhsm               | robinhood_server
 31 | robinhood-webgui             | robinhood_server
 32 | robinhood-adm                | robinhood_server
 95 | mlnx-ofa_kernel-devel        | posix_copytool_worker
 96 | kmod-mlnx-ofa_kernel         | posix_copytool_worker
 97 | mlnx-ofa_kernel-devel        | robinhood_server
 98 | kmod-mlnx-ofa_kernel         | robinhood_server
 99 | foo-package                  | foo
(35 rows)
```

Getting profiles:
image

Removing a profile:
image

Getting profiles again:
image

I will clean up now, and I want to investigate returning ids instead of just bools.

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2363)

### CODE REVIEW

File: iml-api/src/graphql/mod.rs [hunk 1]
```diff
@@ -923,6 +966,138 @@ impl MutationRoot {
...
+            INSERT INTO chroma_core_serverprofilepackage (package_name, server_profile_id)
+            SELECT package_name, $2
+            FROM UNNEST($1::text[])
+            as t(package_name)
+            ON CONFLICT DO NOTHING
+            "#,
+            &profile.packages.into_iter().collect::<Vec<_>>(),
+            &profile.name
+        )
+        .execute(&context.pg_pool)
```
Comment (ip1981) [inline iml-api/src/graphql/mod.rs:1067]: I feel like all these queries should be in a transaction.

File: iml-api/src/graphql/mod.rs [hunk 2]
```diff
@@ -923,6 +966,138 @@ impl MutationRoot {
...
+
+        sqlx::query!(
+            "DELETE FROM chroma_core_serverprofilepackage WHERE server_profile_id = $1",
+            &profile_name
+        )
+        .execute(&context.pg_pool)
+        .await?;
+
+        sqlx::query!(
+            "DELETE FROM chroma_core_serverprofile where name = $1",
```
Comment (ip1981) [inline iml-api/src/graphql/mod.rs:1093]:
```suggestion
            "DELETE FROM chroma_core_serverprofile WHERE name = $1",
```

Comment (ip1981) [inline iml-api/src/graphql/mod.rs:1093]: I think deletting from `chroma_core_serverprofile` will cause deletion from `chroma_core_serverprofile_repolist` and `chroma_core_serverprofilepackage` for the foreing keys. Not sure if we should rely on this.

File: iml-api/src/graphql/mod.rs (hunk 1 above)
Comment (mkpankov) [reply iml-api/src/graphql/mod.rs:1067]: Yep, was just thinking the same.

File: iml-api/src/graphql/mod.rs (hunk 2 above)
Comment (mkpankov) [reply iml-api/src/graphql/mod.rs:1093]: Well, `repolist` is internal Django table for many-to-many relation. `package` should get deleted as it's `ON DELETE CASCADE`. On the other hand, there are `ServerProfileValidation` and `registrationtoken` which aren't mentioned here, but depend on foreign key in `serverprofile`.
I'd say keeping referential integrity is DB's concern. @jgrund ?

Comment (mkpankov) [reply iml-api/src/graphql/mod.rs:1093]: Well, repolist is internal Django table for many-to-many relation. package should get deleted as it's ON DELETE CASCADE. On the other hand, there are ServerProfileValidation and registrationtoken which aren't mentioned here, but depend on foreign key in serverprofile.
I'd say keeping referential integrity is DB's concern. @jgrund ?

File: iml-api/src/graphql/mod.rs [hunk 3]
```diff
@@ -566,6 +567,48 @@ impl QueryRoot {
             },
         })
     }
+
+    async fn server_profiles(context: &Context) -> juniper::FieldResult<ServerProfileResponse> {
+        let server_profile_records = sqlx::query!(
```
Comment (jgrund) [inline iml-api/src/graphql/mod.rs:572]:
```suggestion
        let server_profile_records = sqlx::query_as!(ServerProfileRecord,
```

Could we not just use query_as here?

**Review by jgrund — APPROVED** [review]

File: iml-api/src/graphql/mod.rs (hunk 3 above)
Comment (mkpankov) [reply iml-api/src/graphql/mod.rs:572]: No, we're querying repository data via joins here as well (`jsonb_agg ... AS repos`). We get a struct with all the fields of `ServerProfileRecord` and `repos` field containing `serde_json::Value` with repositories.

Comment (jgrund) [reply iml-api/src/graphql/mod.rs:574]: I wonder if we could cast the item here to it's final type and avoid the need to do it later

**Review by jgrund — APPROVED** [review]

**Review by ip1981 — APPROVED** [review]

**Pull request merged**